src/applications/dynamic_simulation_full_y/dsf_comm_main.cpp

```diff
@@ -0,0 +1,174 @@
+    pfData->getValue("BUS_PF_VANG",&rval);
+    dsData->setValue(BUS_VOLTAGE_ANG,rval);
+    int ngen = 0;
+    if (pfData->getValue(GENERATOR_NUMBER, &ngen)) {
+      int j;
+      for (j=0; j<ngen; j++) {
+        pfData->getValue("GENERATOR_PF_PGEN",&rval,j);
+        dsData->setValue(GENERATOR_PG,rval,j);
+        //printf("save PGEN: %f\n", rval);
+        pfData->getValue("GENERATOR_PF_QGEN",&rval,j);
+        dsData->setValue(GENERATOR_QG,rval,j);
+        //printf("save QGEN: %f\n", rval);
+      }
+    }
+  }
+}
+
+// Calling program for the dynamis simulation applications
+
+void dsf(MPI_Comm comm, int argc, char **argv)
+{
+  gridpack::NoPrint *noprint_ins = gridpack::NoPrint::instance();
+  noprint_ins->setStatus(true);
+  
+  gridpack::Environment env(argc,argv,comm);
+
+  if (env.active()) {
+
+    gridpack::utility::CoarseTimer *timer =
+    gridpack::utility::CoarseTimer::instance();
+    int t_total = timer->createCategory("Dynamic Simulation: Total Application");
+    timer->start(t_total);
+
+    gridpack::parallel::Communicator world;
+
+    // read configuration file 
+    int t_config = timer->createCategory("Dynamic Simulation: Config");
+    timer->start(t_config);
+    gridpack::utility::Configuration *config =
+      gridpack::utility::Configuration::configuration();
+    if (argc >= 2 && argv[1] != NULL) { 
+      char inputfile[256]; 
+      sprintf(inputfile,"%s",argv[1]);
+      config->open(inputfile,world);
+    } else {
+      config->open("input.xml",world);
+    }
+    timer->stop(t_config);
+
+    // setup and run powerflow calculation
+    gridpack::utility::Configuration::CursorPtr cursor;
+    cursor = config->getCursor("Configuration.Powerflow");
+    bool useNonLinear = false;
+    useNonLinear = cursor->get("UseNonLinear", useNonLinear);
+
+    boost::shared_ptr<gridpack::powerflow::PFNetwork>
+      pf_network(new gridpack::powerflow::PFNetwork(world));
+
+    gridpack::powerflow::PFAppModule pf_app;
+    pf_app.readNetwork(pf_network, config);
+    pf_app.initialize();
+    if (useNonLinear) {
+      pf_app.nl_solve();
+    } else {
+      pf_app.solve();
+    }
+    pf_app.write();
+    pf_app.saveData();
+   
+    // setup and run dynamic simulation calculation
+    boost::shared_ptr<gridpack::dynamic_simulation::DSFullNetwork>
+      ds_network(new gridpack::dynamic_simulation::DSFullNetwork(world));
+    gridpack::dynamic_simulation::DSFullApp ds_app;
+    pf_network->clone<gridpack::dynamic_simulation::DSFullBus,
+      gridpack::dynamic_simulation::DSFullBranch>(ds_network);
+
+    // transfer results from PF calculation to DS calculation
+    transferPFtoDS(pf_network, ds_network); 
+
+    // read in faults from input file
+    //gridpack::utility::Configuration::CursorPtr cursor;
+    cursor = config->getCursor("Configuration.Dynamic_simulation");
+    std::vector<gridpack::dynamic_simulation::Event> faults;
+    faults = ds_app.getFaults(cursor);
```

Comment:
Also, `getFaults()` is no longer a method.  I think it's `getEvents()` now. I don't know if there is an internal difference.

Reply:
FIxed.